### DISCUSSION TITLE
Biphasic microstructure

### DISCUSSION THREAD

**Discussion by neelammishra** [neperfepx/neper#465] [Q&A] [answered]
What is the command for generating biphasic microstructure in neper? Can anyone share an example?

**Comment by rquey** [comment, marked as answer]
Material phases can be represented using option [-group](https://neper.info/doc/neper_t.html#cmdoption-group).  For example:
```
neper -T -n 100 -group "id<=50?1:2"
```
will assign all cells of id lower than or equal to 50 to phase 1 and the others to phase 2.

**Reply by neelammishra** [reply]
It is working. However the image has random colours for all cells. Is it possible to assign a single colour to one phase and another color to another phase??

**Reply by rquey** [reply]
Yes, use `-datacellcol group`:
```
neper -V n100-id1.tess -datacellcol group -print img
```
![image](https://user-images.githubusercontent.com/2659376/166876547-decbbb55-c8fb-4815-8357-037d8a96a924.png)

**Reply by neelammishra** [reply]
Thanks. How can we assign volume fraction to the phases??

**Reply by rquey** [reply]
In the example above, changing the number of grains assigned to a phase will simply change the volume fraction of the phase. There is no other option.
